### PR TITLE
Infra: Use Docutils' `list-table` syntax for PEP 0

### PR DESCRIPTION
**Pull request by AA-Turner**
Change in support of subindices.

A

### CODE REVIEW

**Comment by AA-Turner** [comment]
Save for the CLA check, this is now done -- identical output in PEP 0, and a nice reduction in complexity as (almost) all of the custom PEP 0 transform is now gone (only the email address masking logic remains).

A

**Comment by AA-Turner** [comment]
\*Not quite identical output to PEP 0, PEP titles are no longer truncated, and PEP 604's title now links correctly.

A

File: pep_sphinx_extensions/pep_zero_generator/writer.py [hunk 1]
```diff
@@ -80,21 +69,27 @@ def emit_text(self, content: str) -> None:
...
-        self.emit_table_separator()
+        self.emit_text(".. list-table::")
+        self.emit_text("   :header-rows: 1")
+        self.emit_text("   :widths: auto")
+        self.emit_text("   :class: pep-zero-table")
+        self.emit_newline()
+        self.emit_text("   * - ")
+        self.emit_text("     - PEP")
+        self.emit_text("     - PEP Title")
+        self.emit_text("     - PEP Author(s)")
```
Comment (hugovk) [inline pep_sphinx_extensions/pep_zero_generator/writer.py:92]: "Author(s)" is a bit ugly, for a table heading can we just put "Authors" or "Author"?

Also I think it's clear that the title is the _PEP_ title, and the authors are the _PEP_ authors, in a list of PEPs on the PEP index page on https://peps.python.org/ :)

```suggestion
        self.emit_text("     - PEP")
        self.emit_text("     - Title")
        self.emit_text("     - Authors")
```

**Review by hugovk — COMMENTED** [review]
It would be nice to have some tests for the changed `emit_*` functions in `writer.py`. 

(Testing just `emit_pep_category` would also cover `emit_column_headers` and `emit_pep_row`.)

https://codecov.io/gh/python/peps/compare/278070ac9765efbd34e325d6080d8162bd529e99...b3a603d2fede03f207a26eaec584d13e6f8ebb13/src/pep_sphinx_extensions/pep_zero_generator/writer.py?before=pep_sphinx_extensions/pep_zero_generator/writer.py

**Comment by AA-Turner** [comment]
> It would be nice to have some tests

I'm not sure what these would look like -- there's no real point to unit tests of just the function here, as what matters is that the entire list is a valid input to the `list-table` directive (a two-level list where the length of every second level list is uniform).

I'd then argue that the integration test part is handled by if it is parsed correctly as reST!

A

**Comment by AA-Turner** [comment]
(Also, the CLA is still not working -- just tried again)

**Comment by hugovk** [comment]
Yeah, tests not so valuable here. If nothing else, a test would show the code is at least running without exceptions, although we should _probably_ notice it (but if not a super simple test would have helped ;)

CLA: I've pinged @ambv again.

**Pull request merged**